src/pages/scheduler/shared/ical.ts

```diff
@@ -0,0 +1,85 @@
+import { format } from 'date-fns';
+
+const VERSION = 'VERSION:2.0';
+const PROD_ID = 'PRODID:-//courseup//courseup.vikelabs.ca//EN';
+
+export const formatDate = (date: Date) => format(date, "yyyyMMdd'T'kkmmss");
+
+const toDatetime = (date: Date, tz: string, attr: string) => `${attr};TZID=${tz}:${formatDate(date)}`;
+
+type vEvent = {
+  // The following are REQUIRED, but MUST NOT occur more than once.
+  uid: string;
+  dtstamp?: Date;
+  // The following is REQUIRED if the component
+  // appears in an iCalendar object that doesn't
+  // specify the "METHOD" property; otherwise, it
+  // is OPTIONAL; in any case, it MUST NOT occur
+  // more than once.
+  dtstart: Date | string;
```

Comment:
if this is potentially optional should it have ?

Reply:
Yeah I mixed up `dtstamp` and `dtstart`.

Reply:
I added a note to why `dtstamp` can be undefined and fixed `dtstart`